### PR TITLE
binarylogs: split peer address and port into separate fields

### PR DESCRIPTION
**Pull request by zpencer**
Using sockaddr_in is leaking C implementation details into the proto.
Since we really only care about address and port, let's just store
exactly that. This makes it much easier for tooling and non-C gRPC
implementations to deal with Peer.

### CODE REVIEW

File: grpc/binlog/v1alpha/binarylog.proto [hunk 1]
```diff
@@ -97,12 +97,13 @@ message MetadataEntry {
 message Peer {
   enum PeerType {
     UNKNOWN_PEERTYPE = 0;
-    PEER_IPV4 = 1;  // peer is struct sockaddr_in
-    PEER_IPV6 = 2;  // peer is struct sockaddr_in6
-    PEER_UNIX = 3;  // peer is struct sockaddr_un
+    PEER_IPV4 = 1;  // peer is 4 byte ipv4 address
```
Comment (yashykt) [inline grpc/binlog/v1alpha/binarylog.proto]: That won't be human readable

Comment (zpencer) [reply grpc/binlog/v1alpha/binarylog.proto]: I'm all for making things more readable. What do you think about using the [canonical ipv6](https://tools.ietf.org/html/rfc5952#section-4) address format?

**Comment by zpencer** [comment]
@yashykt PTAL. The ipv6 scope is only  applicable for link local addresses, I don't think it's a priority to support it today but it will exist here for completeness.

**Comment by zpencer** [comment]
uint16 is not a valid proto field type.
use a new field for "address" to make it easier to migrate to this new proto.

**Pull request merged**